tests/integration_tests/functional/test_snapshot_basic.py

```diff
@@ -351,6 +352,82 @@ def test_5_inc_snapshots(network_config,
 
     test_matrix.run_test(_test_seq_snapshots)
 
+DEFAULT_HOST_IP = "192.168.0.1"
+DEFAULT_GUEST_IP = "192.168.0.2"
+DEFAULT_TAP_NAME = "tap0"
+DEFAULT_DEV_NAME = "eth0"
+DEFAULT_NETMASK = 30
+
+def test_load_snapshot_failure_handling(bin_cloner_path):
+    """Test scenario:
+    1.  Create a new snapshot from a Firecracker process.
+    2.  Corrupt the snapshot's memory file.
+    3.  Try to build a VM out of the corrupted snapshot.
+    4.  Verify that an error was shown and the process is dead. """
+
+    logger = logging.getLogger("snapshot_load_failure")
+
+    # Use a predefined vm instance.
+    vm_instance = VMNano.spawn(bin_cloner_path)
+    vm = vm_instance.vm
+    root_disk = vm_instance.disks[0]
+    ssh_key = vm_instance.ssh_key
+
+    vm.start()
+    ssh_connection = net_tools.SSHConnection(vm.ssh_config)
+
+    # Verify if guest can run commands.
+    exit_code, _, _ = ssh_connection.execute_command("sync")
+    assert exit_code == 0
+
+    logger.info("Create %s.", SnapshotType.FULL)
+    # Create a snapshot builder from a microvm.
+    snapshot_builder = SnapshotBuilder(vm)
+
+    disks = [root_disk.local_path()]
+    # Create base snapshot.
+    snapshot = snapshot_builder.create(disks,
+                                       ssh_key,
+                                       SnapshotType.FULL)
+
+    vm.kill()
+
+    # Corrupt the memory file of the snapshot
+    logger.info("Corrupt the memory file %s", snapshot.mem)
+
+    mem_fd = open(snapshot.mem, "wb")
+    mem_fd.write(bytearray([0xde, 0xad, 0xbe, 0xef, 0x00]))
```

Comment:
How does this guarantee the `LoadSnapshot` API command will fail? Wouldn't it be more robust to just use an empty or completely zeroed out snapshot vm-state file?

Memory corruptions will affect guest and while we do have _some_ validations for it, it is not guaranteed that the snapshot load operation will fail because of it.

Reply:
You are correct. I will empty the file in this case.